### PR TITLE
60 add contacts to concerts

### LINKED ISSUE #60: Add contacts to concerts
Either have them in the fields or create an association class

### PR DESCRIPTION
**Pull request by SKarthikeyan28**
Added ConcertContact class to associate Persons with Concert

### CODE REVIEW

**Review by wes-w-z-h — COMMENTED** [review]
How to add a contact to the concert.
Need a command for that or update an existing one

File: src/main/java/seedu/address/model/concert/ConcertContact.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        requireAllNonNull(person, concert);
+
+        this.person = person;
+        this.concert = concert;
+
+        addContactToConcert(this); //instantly adds Person to Concert list as ConcertContact created
+    }
+
+    public Person getPerson() {
+        return this.person;
```
Comment (wes-w-z-h) [inline src/main/java/seedu/address/model/concert/ConcertContact.java]: no need for this.person, person will do since unambiguous in this case

File: src/main/java/seedu/address/model/concert/ConcertContact.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+    public Concert getConcert() {
+        return this.concert;
```
Comment (wes-w-z-h) [inline src/main/java/seedu/address/model/concert/ConcertContact.java]: same here

File: src/main/java/seedu/address/model/concert/ConcertContact.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof ConcertContact)) {
+            return false;
+        }
+
+        ConcertContact otherContact = (ConcertContact) other;
+        return otherContact.person.equals(person)
```
Comment (wes-w-z-h) [inline src/main/java/seedu/address/model/concert/ConcertContact.java:74]: should do a null check as a sanity check in case otherContact is null

File: src/main/java/seedu/address/model/concert/ConcertContact.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    private static void addContactToConcert(ConcertContact contact) {
+        concertContacts.computeIfAbsent(contact.getConcert(), key -> new ArrayList<>()).add(contact);
+    }
+
+    /**
+     * Returns the list of ConcertContacts in the Concert
+     *
+     * @return List of ConcertContact
+     */
+    public ArrayList<ConcertContact> concertContactList(Concert concert) {
```
Comment (wes-w-z-h) [inline src/main/java/seedu/address/model/concert/ConcertContact.java]: this should be named: getConcertContactList or something since it is a getter, also should return an immutable object instead of the array list directly

**Review by wes-w-z-h — CHANGES_REQUESTED** [review]
Slight changes
can consider adding more test cases like adding to an existing concert/ negative test cases

File: src/main/java/seedu/address/model/concert/ConcertContact.java (hunk 3 above)
Comment (wes-w-z-h) [reply src/main/java/seedu/address/model/concert/ConcertContact.java:74]: ignore this instanceof handles null

File: src/main/java/seedu/address/model/concert/ConcertContact.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        concertContacts.computeIfAbsent(contact.getConcert(), key -> new ArrayList<>()).add(contact);
+    }
+
+    /**
+     * Returns the list of ConcertContacts in the Concert
+     *
+     * @return List of ConcertContact
+     */
+    public ArrayList<ConcertContact> getConcertContactList(Concert concert) {
+        return concertContacts.get(concert);
```
Comment (wes-w-z-h) [inline src/main/java/seedu/address/model/concert/ConcertContact.java]: this is mutable can consider returning an unmodifiable list

File: src/main/java/seedu/address/model/concert/ConcertContact.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof ConcertContact)) {
+            return false;
+        }
+
+        ConcertContact otherContact = (ConcertContact) other;
+        return otherContact != null && otherContact.person.equals(person)
```
Comment (wes-w-z-h) [inline src/main/java/seedu/address/model/concert/ConcertContact.java]: actually dont need this null check my bad

File: src/test/java/seedu/address/model/concert/ConcertContactTest.java
Comment (wes-w-z-h) [inline src/test/java/seedu/address/model/concert/ConcertContactTest.java:1]: can add an equals test

File: src/main/java/seedu/address/model/concert/ConcertContact.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                && otherContact.concert.equals(concert);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(person, concert);
+    }
+
+    @Override
+    public String toString() {
```
Comment (wes-w-z-h) [inline src/main/java/seedu/address/model/concert/ConcertContact.java:84]: may have issues later on but for now can leave it

**Review by wes-w-z-h — CHANGES_REQUESTED** [review]

**Review by wes-w-z-h — APPROVED** [review]
lgtm!
